ffi/firewood_test.go

```diff
@@ -341,7 +341,311 @@ func TestFakeProposal(t *testing.T) {
 	_, err := proposal.Get([]byte("non-existent"))
 	require.Contains(t, err.Error(), "proposal not found", "Get(fake proposal)")
 
+	// Attempt to create a new proposal from the fake proposal.
+	_, err = proposal.Propose([][]byte{[]byte("key")}, [][]byte{[]byte("value")})
+	require.Contains(t, err.Error(), "proposal not found", "Propose(fake proposal)")
+
 	// Attempt to commit the fake proposal.
 	err = proposal.Commit()
 	require.Contains(t, err.Error(), "proposal not found", "Commit(fake proposal)")
 }
+
+// Tests that a proposal with an invalid ID can be dropped.
+func TestDropFakeProposal(t *testing.T) {
+	db := newTestDatabase(t)
+
+	// Create a fake proposal with an invalid ID.
+	proposal := &Proposal{
+		handle: db.handle,
+		id:     1, // note that ID 0 is reserved for invalid proposals
+	}
+
+	// Attempt to drop the fake proposal.
+	// This should not be an error, since dropping a proposal is a "no-op".
```

Comment:
This comment seems wrong. It is an error, as the proposal ID being passed in is not valid.